packages/core/src/common/cluster/request-namespace-list-permissions.test.ts

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { V1SubjectRulesReviewStatus } from "@kubernetes/client-node";
+import type { DiContainer } from "@ogre-tools/injectable";
+import { getDiForUnitTesting } from "../../main/getDiForUnitTesting";
+import type { RequestNamespaceListPermissionsFor } from "./request-namespace-list-permissions.injectable";
+import requestNamespaceListPermissionsForInjectable from "./request-namespace-list-permissions.injectable";
+
+const createFakeProxyConfig = (statusResponse: Promise<{ body: { status: V1SubjectRulesReviewStatus }}>) => ({
+  makeApiClient: () => ({
+    createSelfSubjectRulesReview: (): Promise<{ body: { status: V1SubjectRulesReviewStatus }}> => statusResponse,
+  }),
+});
+
+describe("requestNamespaceListPermissions", () => {
+  let di: DiContainer;
+  let requestNamespaceListPermissions: RequestNamespaceListPermissionsFor;
+
+  beforeEach(() => {
+    di = getDiForUnitTesting({ doGeneralOverrides: true });
+    requestNamespaceListPermissions = di.inject(requestNamespaceListPermissionsForInjectable);
+  });
+
+  describe("when api returns incomplete data", () => {
+    it("returns truthy function", async () => {
+      const requestPermissions = requestNamespaceListPermissions(createFakeProxyConfig(
+        new Promise((resolve) => resolve({
+          body: {
+            status: {
+              incomplete: true,
+              resourceRules: [],
+              nonResourceRules: [],
+            },
+          },
+        })),
```

Comment:
Using `asyncFn` for stuff like this would remove a bit of duplication :)

Reply:
Maybe give an example and I'll refactor? :)

Reply:
An example on adding `asyncFn` to existing code structure. Note that the different parts of code here can be spread in different `beforeEach` to remove said duplication.

```typescript
const createSelfSubjectRulesReviewMock = asyncFn();

const proxyConfigStub = createProxyConfigStub(createSelfSubjectRulesReviewMock);

const requestPermissions = requestNamespaceListPermissions(proxyConfigStub);

const permissionCheckPromise = requestPermissions("some-namespace");

// Note: in different test, this could be `.reject(...)`.
createSelfSubjectRulesReviewMock.resolve({
  body: {
    status: {
      incomplete: true,
      resourceRules: [],
      nonResourceRules: [],
    },
  },
});

const permissionCheck = await permissionCheckPromise;

expect(permissionCheck({
  apiName: "pods",
  group: "",
  kind: "Pod",
  namespaced: true,
})).toBeTruthy();
```